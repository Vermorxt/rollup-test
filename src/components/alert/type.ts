import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
