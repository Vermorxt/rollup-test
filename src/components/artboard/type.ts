import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
