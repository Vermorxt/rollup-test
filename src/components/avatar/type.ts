import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
