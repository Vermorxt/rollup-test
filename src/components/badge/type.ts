import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
