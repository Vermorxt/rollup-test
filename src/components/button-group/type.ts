import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
