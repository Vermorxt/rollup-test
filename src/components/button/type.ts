import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
