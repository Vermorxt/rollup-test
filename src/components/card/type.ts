import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
