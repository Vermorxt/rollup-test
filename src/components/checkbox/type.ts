import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
export type CheckboxPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Checkbox_variants {
  disabled?: boolean
}

export interface Ui_Checkbox_singleClass {
  loading?: boolean
  checked?: boolean
  indeterminate?: boolean
  label?: string
  labelPosition?: 'left' | 'right'
}

export interface Ui_CheckboxProps
  extends HTMLProps<HTMLInputElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Checkbox_variants,
    Ui_Checkbox_singleClass {
  position?: CheckboxPositionProps
}
