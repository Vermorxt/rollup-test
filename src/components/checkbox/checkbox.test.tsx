import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { itHasCssClass } from '../../_tests-utils/it-has-color-css-class'
import { checkAccessibility, itSupportsSystemProps } from '../../_tests-utils'
import { Ui_CheckboxProps } from './type'
import Ui_Checkbox from './checkbox'
import React from 'react'

const defaultProps: Ui_CheckboxProps = {}

const selector = 'text'

describe('@cms/core/Ui_Checkbox', () => {
  checkAccessibility([<Ui_Checkbox {...defaultProps} label="test" ref={null} />])
  itSupportsSystemProps({
    component: Ui_Checkbox,
    props: defaultProps,
    refType: HTMLInputElement,
  })

  // NOTE: (possible debug): debugComponent(<Ui_Checkbox {...defaultProps} ref={null} />)

  it('should match snapshot', () => {
    const { container } = render(<Ui_Checkbox {...defaultProps} ref={null} />)
    expect(container).toMatchSnapshot()
  })

  it('renders SIZE classes', async () => {
    await itHasCssClass(<Ui_Checkbox large label="x" />, `.${selector}-xl`)
    await itHasCssClass(<Ui_Checkbox medium label="x" />, `.${selector}-base`)
    await itHasCssClass(<Ui_Checkbox small label="x" />, `.${selector}-sm`)
    await itHasCssClass(<Ui_Checkbox tiny label="x" />, `.${selector}-xs`)
    await itHasCssClass(<Ui_Checkbox mini label="x" />, `.${selector}-xxs`)
  })

  it('should be disabled', async () => {
    const labelText = 'label-text'

    const { container: checkbox } = render(<Ui_Checkbox label={labelText} disabled={true} ref={null} />)

    const label = screen.getByLabelText(labelText)
    expect(label).toBeDisabled()
  })

  it('should have checked state on label click', async () => {
    let checkboxChecked = false
    const mockCallback = jest.fn(e => (checkboxChecked = e?.target?.checked))

    const labelText = 'label-text'
    const { container: checkbox } = render(
      <Ui_Checkbox {...defaultProps} checked={checkboxChecked} label={labelText} ref={null} onChange={mockCallback} />
    )

    expect(checkbox).not.toBeChecked()

    const label = screen.getByLabelText(labelText)
    await fireEvent.click(label)

    expect(mockCallback).toHaveBeenCalled()
    expect(checkboxChecked).toBeTruthy()
  })

  it('renders bg color classes', async () => {
    await itHasCssClass(<Ui_Checkbox primary />, `.checkbox-primary`)
    await itHasCssClass(<Ui_Checkbox secondary />, `.checkbox-secondary`)
    await itHasCssClass(<Ui_Checkbox accent />, `.checkbox-accent`)
    await itHasCssClass(<Ui_Checkbox info />, `.checkbox-info`)
    await itHasCssClass(<Ui_Checkbox error />, `.checkbox-error`)
    await itHasCssClass(<Ui_Checkbox success />, `.checkbox-success`)
    await itHasCssClass(<Ui_Checkbox warning />, `.checkbox-warning`)
    await itHasCssClass(<Ui_Checkbox ghost />, `.checkbox-ghost`)
    await itHasCssClass(<Ui_Checkbox neutral />, `.checkbox-neutral`)
    await itHasCssClass(<Ui_Checkbox link />, `.checkbox-link`)
  })
})
