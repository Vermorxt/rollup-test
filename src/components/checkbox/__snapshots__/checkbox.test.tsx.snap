// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Checkbox should match snapshot 1`] = `
<div>
  <input
    class="
      checkbox 
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
      
      "
    type="checkbox"
  />
</div>
`;
